go/vt/vtgate/evalengine/compare.go

```diff
@@ -121,17 +121,17 @@ func compareNumeric(left, right eval) (int, error) {
 //   - https://dev.mysql.com/doc/refman/8.0/en/type-conversion.html
 //   - https://dev.mysql.com/doc/refman/8.0/en/date-and-time-type-conversion.html
 func compareDates(l, r *evalTemporal) (int, error) {
-	return compareGoTimes(l.toStdTime(), r.toStdTime())
+	return compareGoTimes(l.toStdTime(time.Local), r.toStdTime(time.Local))
```

Comment:
I think we can avoid having to convert to `time.Time` here and use `DateTime#Compare` here?